options/posix/generic/pthread-stubs.cpp

```diff
@@ -923,10 +923,10 @@ int pthread_cond_wait(pthread_cond_t *__restrict cond, pthread_mutex_t *__restri
 	return 0;
 }
 
-int pthread_cond_timedwait(pthread_cond_t *__restrict, pthread_mutex_t *__restrict,
+int pthread_cond_timedwait(pthread_cond_t *__restrict cond, pthread_mutex_t *__restrict mutex,
 		const struct timespec *__restrict) {
-	__ensure(!"Not implemented");
-	__builtin_unreachable();
+	// This is wrong, needs changing
+	return pthread_cond_wait(cond, mutex);
```

Comment:
Hm, this is indeed wrong :smile:. I think it should not be too hard to change it, we essentially need to pass a timeout to the `sys_futex_wait` call in pthread_cond_wait.